platform/ios/src/MGLMapView.mm

```diff
@@ -1421,10 +1418,47 @@ - (void)actionSheet:(UIActionSheet *)actionSheet didDismissWithButtonIndex:(NSIn
         [[UIApplication sharedApplication] openURL:
          [NSURL URLWithString:feedbackURL]];
     }
-    // skips to 4 because button is conditionally added after cancel (index 3)
-    else if (buttonIndex == actionSheet.firstOtherButtonIndex + 4)
+    else if (buttonIndex == actionSheet.firstOtherButtonIndex + 3)
+    {
+        NSString *messageTense;
+        NSString *participate;
+        NSString *optOut;
+
+        if ([[NSUserDefaults standardUserDefaults] boolForKey:@"MGLMapboxMetricsEnabled"])
+        {
+            messageTense = @"are helping to";
+            participate = @"Keep Participating";
+            optOut = @"Stop Participating";
+        }
+        else
+        {
+            messageTense = @"can help";
+            participate = @"Participate";
+            optOut = @"Don’t Participate";
+        }
+
+        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Make Mapbox Maps Better"
+                                                        message:[NSString stringWithFormat:@"You %@ make OpenStreetMap and Mapbox maps better by contributing anonymous usage data.", messageTense]
+                                                       delegate:self
+                                              cancelButtonTitle:participate
+                                              otherButtonTitles:@"Tell Me More", optOut, nil];
+        [alert show];
+    }
+}
+
+- (void)alertView:(UIAlertView *)alertView didDismissWithButtonIndex:(NSInteger)buttonIndex
+{
+    if (buttonIndex == alertView.cancelButtonIndex)
```

Comment:
A `switch` requires constant values (`Case value is not a constant expression`), which here would mean hardcoding the indexes rather than using `alertView.cancelButtonIndex` and so on. Is that what you had in mind?